src/main/java/run/halo/sticker/endpoint/StickerAttachmentEndpoint.java

```diff
@@ -0,0 +1,190 @@
+
+import com.google.common.io.Files;
+import java.time.Duration;
+import java.util.Date;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.core.io.buffer.DataBuffer;
+import org.springframework.dao.OptimisticLockingFailureException;
+import org.springframework.http.MediaType;
+import org.springframework.http.codec.multipart.FilePart;
+import org.springframework.http.codec.multipart.Part;
+import org.springframework.security.core.context.ReactiveSecurityContextHolder;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.stereotype.Component;
+import org.springframework.util.MultiValueMap;
+import org.springframework.util.unit.DataSize;
+import org.springframework.web.reactive.function.BodyExtractors;
+import org.springframework.web.reactive.function.server.RouterFunction;
+import org.springframework.web.reactive.function.server.ServerRequest;
+import org.springframework.web.reactive.function.server.ServerResponse;
+import org.springframework.web.server.ServerWebInputException;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.util.retry.Retry;
+import run.halo.app.core.extension.attachment.Attachment;
+import run.halo.app.core.extension.endpoint.CustomEndpoint;
+import run.halo.app.core.extension.service.AttachmentService;
+import run.halo.app.extension.GroupVersion;
+import run.halo.app.extension.Metadata;
+import run.halo.app.extension.ReactiveExtensionClient;
+import run.halo.app.plugin.ReactiveSettingFetcher;
+import run.halo.sticker.infra.StickerSetting;
+import run.halo.sticker.model.Sticker;
+import run.halo.sticker.model.StickerGroup;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class StickerAttachmentEndpoint implements CustomEndpoint {
+
+    private static final String SELF_USER = "-";
+    private static final String STICKER_GROUP_NAME = "sticker-group";
+    private static final String DEFAULT_STICKER_ATTACHMENT_POLICY_NAME = "default-policy";
+    private static final DataSize MAX_FILE_SIZE = DataSize.ofMegabytes(2L);
+    private final ReactiveExtensionClient client;
+    private final AttachmentService attachmentService;
+    private final ReactiveSettingFetcher settingFetcher;
+
+    @Override
+    public RouterFunction<ServerResponse> endpoint() {
+        var tag = "StickerV1alpha1Console";
+        return route()
+            // a get request resp test success text
+            .GET("sticker/test", this::test)
+            .POST("sticker/{groupName}/upload", contentType(MediaType.MULTIPART_FORM_DATA),
+                this::uploadUserSticker)
+            .build();
+    }
+
+    @Override
+    public GroupVersion groupVersion() {
+        return CustomEndpoint.super.groupVersion();
```

Comment:
使用具体的 `groupVersion`，例如 `console.api.sticker.halo.run/v1alpha1`

```suggestion
        return GroupVersion.parseAPIVersion("console.api.sticker.halo.run/v1alpha1`");
```